webview-ui/src/components/settings/constants.ts

```diff
@@ -36,6 +36,7 @@ export const PROVIDERS = [
 	{ value: "unbound", label: "Unbound" },
 	{ value: "requesty", label: "Requesty" },
 	{ value: "human-relay", label: "Human Relay" },
+	{ value: "fake-ai", label: "Fake AI" },
```

Comment:
Should we hide this in the UI?

Reply:
I think it is better to hide it. My only concern would be if the `fake-ai` provider is set via `RooCodeAPI.setConfiguration`, and later the settings view is opened as part of the e2e test, wouldn't this lead to some inconsistent state and possibly the setting view "normalizing" and overwriting the config value? Although, you could just say "don't open settings during e2e tests" and that would be fine for us, since we can use setConfiguration to setup settings for a test.

Reply:
Fixed.
It seems that when Fake AI is set as an API provider, in the settings view the API Provider input simply shows an empty field (no API provider is selected).
Saving the settings will clear the `fakeAi` object that handles API requests (and thus further requests will fail until the user sets another API provider). Showing settings view without saving does not overwrite configuration.